kernel/src/hil/framebuffer.rs

```diff
@@ -0,0 +1,143 @@
+//! Interface for FrameBuffer
+use crate::returncode::ReturnCode;
+use enum_primitive::cast::FromPrimitive;
+use enum_primitive::enum_from_primitive;
+
+enum_from_primitive! {
+    #[derive(Copy, Clone, PartialEq)]
+    pub enum ScreenRotation {
+        Normal = 0,
+        Rotated90 = 1,
+        Rotated180 = 2,
+        Rotated270 = 3,
+    }
+}
+
+impl From<ScreenRotation> for usize {
+    fn from(rotation: ScreenRotation) -> usize {
+        match rotation {
+            ScreenRotation::Normal => 0,
+            ScreenRotation::Rotated90 => 1,
+            ScreenRotation::Rotated180 => 2,
+            ScreenRotation::Rotated270 => 3,
+        }
+    }
+}
+
+enum_from_primitive! {
+    #[derive(Copy, Clone, PartialEq)]
+    #[allow(non_camel_case_types)]
+    pub enum ScreenColorDepth {
+        None = 0,
+        Mono = 1,
+        RGB_233 = 8,
+        RGB_565 = 16,
+        RGB_888 = 24,
+        ARGB_8888 = 32,
+    }
+}
+
+impl From<ScreenColorDepth> for usize {
+    fn from(color_depth: ScreenColorDepth) -> usize {
+        match color_depth {
+            ScreenColorDepth::None => 0,
+            ScreenColorDepth::Mono => 1,
+            ScreenColorDepth::RGB_233 => 8,
+            ScreenColorDepth::RGB_565 => 16,
+            ScreenColorDepth::RGB_888 => 16,
+            ScreenColorDepth::ARGB_8888 => 32,
+        }
+    }
+}
+
+pub trait Screen {
+    /// Sets the screen resolution (in pixels). Returns ENOSUPPORT if the resolution is
+    /// not supported. The function should return SUCCESS for at least one resolution.
+    fn set_resolution(&self, width: usize, height: usize) -> ReturnCode;
+
+    /// Sets the color depth (in bits per pixel). Returns ENOSUPPORT if the color depth is
+    /// not supported. The function should return SUCCESS for at least one color depth.
+    fn set_color_depth(&self, depth: ScreenColorDepth) -> ReturnCode;
+
+    /// Sets the rotation of the display.
+    /// note this can swap the width with height.
+    fn set_rotation(&self, rotation: ScreenRotation) -> ReturnCode;
+
+    /// Returns a tuple (width, height) with the current resolution (in pixels)
+    /// note that width and height may change due to rotation
+    /// This function is synchronous.
+    fn get_resolution(&self) -> (usize, usize);
```

Comment:
Should `set_resolution ` be updated to accept a tuple? Bit weird that these two don't match.

Reply:
Fixed